operator/internal/manifests/internal/alerts/prometheus-alerts.yaml

```diff
@@ -82,7 +70,114 @@ groups:
         increase(
           loki_panic_total[10m]
         )
-      ) by (namespace, job)
+      ) by (job, namespace)
       > 0
     labels:
-        severity: critical
+      severity: critical
+  - alert: LokiRequestLatency
+    annotations:
+      message: |-
+        {{ $labels.job }} {{ $labels.route }} is experiencing {{ printf "%.2f" $value }}s 99th percentile latency.
+      summary: "The 99th percentile is experiencing high latency (higher than 1 second)."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Request-Latency"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          irate(
+            loki_request_duration_seconds_bucket{route!~"(?i).*tail.*"}[1m]
+          )
+        ) by (le, job, namespace, route)
+      )
+      * 100
+      > 1
+    for: 15m
+    labels:
+      severity: critical
+  - alert: LokiTenantRateLimit
+    annotations:
+      message: |-
+        {{ $labels.job }} {{ $labels.route }} is experiencing 429 errors.
+      summary: "At least 10% of requests are responded with the rate limit error code."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Tenant-Rate-Limit"
+    expr: |
+      sum(
+        code:loki_request_duration_seconds_count:irate1m{status_code="429"}
+      ) by (job, namespace, route)
+      /
+      sum(
+        code:loki_request_duration_seconds_count:irate1m
+      ) by (job, namespace, route)
+      * 100
+      > 10
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiStorageSlowWrite
+    annotations:
+      message: |-
+        The storage path is experiencing slow write response rates.
+      summary: "The storage path is experiencing slow write response rates."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Storage-Slow-Write"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          operation:loki_boltdb_shipper_request_duration_seconds_bucket:rate5m{operation="WRITE"}
+        ) by (le, job, namespace)
+      )
+      * 100
+      > 1
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiStorageSlowRead
+    annotations:
+      message: |-
+        The storage path is experiencing slow read response rates.
+      summary: "The storage path is experiencing slow read response rates."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Storage-Slow-Read"
+    expr: |
+      histogram_quantile(0.99,
+        sum(
+          operation:loki_boltdb_shipper_request_duration_seconds_bucket:rate5m{operation="Shipper.Query"}
+        ) by (le, job, namespace)
+      )
+      * 100
+      > 5
+    for: 15m
+    labels:
+      severity: warning
+  - alert: LokiWritePathHighLoad
+    annotations:
+      message: |-
+        The write path is experiencing high load.
+      summary: "The write path is experiencing high load, causing backpressure storage flushing."
+      runbook_url: "[[ .RunbookURL ]]#Loki-Write-Path-High-Load"
+    expr: |
+      sum(
+        max_over_time(
+          loki_ingester_wal_replay_flushing[1m]
+        )
```

Comment:
I don't think we need the `max_over_time` here. Just using the raw number should be fine. The "over time" part should be handled by the `for: 15m` clause of the alert.